frame/collective/src/lib.rs

```diff
@@ -704,10 +727,131 @@ decl_module! {
+				Ok(origin) => origin,
+				Err(_) => return DispatchError::BadOrigin.into(),
+			};
+
+			let proposal_len = proposal.using_encoded(|x| x.len());
+			ensure!(proposal_len <= length_bound as usize, Error::<T, I>::WrongProposalLength);
+
+			let account = Self::origin_to_ratio_account(raw_origin, target_n, target_d)?;
+			let proposal_hash = T::Hashing::hash_of(&proposal);
+			let signed_origin = frame_system::RawOrigin::Signed(account.clone());
+			let result = proposal.dispatch(signed_origin.into());
+			Self::deposit_event(
+				RawEvent::AccountExecuted(account, proposal_hash, result.map(|_| ()).map_err(|e| e.error))
+			);
+			Ok(())
+		}
+
+		/// Re-dispatch a collective origin call as a unique account that represents that origin.
+		///
+		/// Since the collective may want to be able to represent a different ratio than exists,
+		/// we allow the re-dispatch to specify the target ratio, as long as it is less restrictive
+		/// than the provided origin.
+		///
+		/// Can only be called by the collective origin.
+		#[weight = {
+			T::WeightInfo::dispatch_as_quantity_account(*length_bound)
+			.saturating_add(proposal.get_dispatch_info().weight)
+		}]
+		fn dispatch_as_quantity_account(origin,
+			proposal: Box<T::Proposal>,
+			#[compact] length_bound: u32,
+		) -> DispatchResult {
+			let maybe_raw_origin: Result<RawOrigin<T::AccountId, I>, _> = <T as Config<I>>::Origin::from(origin).into();
+
+			let raw_origin = match maybe_raw_origin {
+				Ok(origin) => origin,
+				Err(_) => return DispatchError::BadOrigin.into(),
+			};
+
+			let proposal_len = proposal.using_encoded(|x| x.len());
+			ensure!(proposal_len <= length_bound as usize, Error::<T, I>::WrongProposalLength);
+
+			let account = Self::origin_to_quantity_account(raw_origin)?;
+			let proposal_hash = T::Hashing::hash_of(&proposal);
+			let signed_origin = frame_system::RawOrigin::Signed(account.clone());
+			let result = proposal.dispatch(signed_origin.into());
+			Self::deposit_event(
+				RawEvent::AccountExecuted(account, proposal_hash, result.map(|_| ()).map_err(|e| e.error))
+			);
+			Ok(())
+		}
 	}
 }
 
 impl<T: Config<I>, I: Instance> Module<T, I> {
+	/// A simple helper function to generate a collective ratio account from the collective origin.
+	///
+	/// A ratio account is an account which represents a ratio of n / d users where n is the number
+	/// of approving members, and d is the total number of members. n and d can be any fraction which
+	/// is less or equally restrictive than the origin.
+	pub fn origin_to_ratio_account(
+		origin: RawOrigin<T::AccountId, I>,
+		target_n: u32,
+		target_d: u32,
+	) -> Result<T::AccountId, DispatchError> {
+		match origin {
+			RawOrigin::Members(n, d) => {
+				// n is the number of members who voted aye. d is the total number of members.
+				// Should never be able to make an origin which represents more members than exists.
+				ensure!(target_d <= d, Error::<T, I>::NotEnoughMembers);
```

Comment:
Is this necessary? If you have a collective of 10 members, you can have a 1/11 origin where a single member fulfills the requirements to assume this origin. Likewise an 11/12 origin would resolve to unanimous.

Reply:
I think it is necessary. For you to create a 1/11 origin, there should be 11 members in the membership.

The thing we want to combat is something like a  1 / 2 origin trying to turn itself into a 4 / 8 origin. They are both one-half, but 4 / 8 origin has more power.

I think checking either `target_d <= d` or `target_n <= n` or both would be sufficient here.

Reply:
> The thing we want to combat is something like a 1 / 2 origin trying to turn itself into a 4 / 8 origin. They are both one-half, but 4 / 8 origin has more power.

I don't totally follow. A 4/8 has more power because it requires 4 affirmatives, even if the collective has e.g. 3 members (in which case, 2-of-3 would satisfy a 1/2 origin but not a 4/8 origin)?

I'm trying to balance this with the fact that collectives will grow. Kusama Council size already increased, and tech committees will as well. So now (with 3 members) a 3/4 TechComm is unanimous, but if (when) there are 5 members, 3/4 would be more meaningful.

These are all a bit edge-casey. I think most paras will work with the least reduction, like 1/2, 2/3, 3/4, 3/5.

Reply:
No. 4/8 origin can satisfy and create a 1/2 account.

But a 1/2 should not be able to satisfy and create a 4/8 account.

The check is ensuring that if I come with `Members(n, d)` I cannot create `Account(target_n/target_d)` where `target_d <= d`

Reply:
I see. Something that I will put in the notes about launching Statemint. Basically, use the reduced ratio in 99% of cases, and only non-reduced ratios like 4/8 if you have a special need, and you will also need to monitor changes to collective sizes on the Relay Chain (and upgrade accordingly).

Reply:
Just to add to all of this, I think i wrote in the comments somewhere.

Statemint we probably want to just program in 1/2 origin as Sudo or whatever.

Council on Polakdot right now is like 13 people, so it would be impossible to automatically reduce that fraction to exactly 1/2. So the point is that 7/13 membership can create a 1/2 account, and even if we change the council size later, they will still be able to create the appropriate account type.